File: core/src/nl/t64/game/rpg/screens/world/conversation/ConversationObserver.java
package nl.t64.game.rpg.screens.world.conversation;

import nl.t64.game.rpg.components.loot.Loot;


public interface ConversationObserver {

    default void onNotifyExitConversation() {
        throw new IllegalCallerException("Implement this method in child.");
    }

    default void onNotifyShowMessageTooltip(String message) {
        throw new IllegalCallerException("Implement this method in child.");
    }

    default void onNotifyShowLevelUpDialog(String message) {
        throw new IllegalCallerException("Implement this method in child.");
    }

    default void onNotifyLoadShop() {
        throw new IllegalCallerException("Implement this method in child.");
    }

    default void onNotifyShowRewardDialog(Loot reward, String levelUpMessage) {
        throw new IllegalCallerException("Implement this method in child.");
    }

    default void onNotifyShowReceiveDialog(Loot receive) {
        throw new IllegalCallerException("Implement this method in child.");
    }

    default void onNotifyHeroJoined() {
        throw new IllegalCallerException("Implement this method in child.");
    }

    default void onNotifyHeroDismiss() {
        throw new IllegalCallerException("Implement this method in child.");
    }

}

File: core/src/nl/t64/game/rpg/screens/world/conversation/ConversationObserver.kt
package nl.t64.game.rpg.screens.world.conversation

import nl.t64.game.rpg.components.loot.Loot


interface ConversationObserver {

    fun onNotifyExitConversation()

    fun onNotifyShowMessageTooltip(message: String) {
        throw IllegalStateException("Implement this method in child.")
    }

    fun onNotifyShowLevelUpDialog(message: String) {
        throw IllegalStateException("Implement this method in child.")
    }

    fun onNotifyLoadShop() {
        throw IllegalStateException("Implement this method in child.")
    }

    fun onNotifyShowRewardDialog(reward: Loot, levelUpMessage: String?) {
        throw IllegalStateException("Implement this method in child.")
    }

    fun onNotifyShowReceiveDialog(receive: Loot) {
        throw IllegalStateException("Implement this method in child.")
    }

    fun onNotifyHeroJoined() {
        throw IllegalStateException("Implement this method in child.")
    }

    fun onNotifyHeroDismiss() {
        throw IllegalStateException("Implement this method in child.")
    }

}
